megamek/src/megamek/client/ui/swing/util/UIUtil.java

```diff
@@ -409,7 +411,169 @@ public static void scaleMenu(final JComponent popup) {
             } 
         }
     }
-    
+
+
+    /**
+     *
+     * @param currentMonitor
+     * @return the width of the screen taking into account display scaling
+     */
+    public static int getScaledScreenWidth(DisplayMode currentMonitor) {
+        int monitorW = currentMonitor.getWidth();
+        int pixelPerInch= Toolkit.getDefaultToolkit().getScreenResolution();
+        return DEFAULT_DISPLAY_PPI * monitorW / pixelPerInch;
+    }
+
+    /**
+     *
+     * @param currentMonitor
+     * @return The height of the screen taking into account display scaling
+     */
+    public static int getScaledScreenHeight(DisplayMode currentMonitor) {
+        int monitorH = currentMonitor.getHeight();
+        int pixelPerInch= Toolkit.getDefaultToolkit().getScreenResolution();
+        return DEFAULT_DISPLAY_PPI * monitorH / pixelPerInch;
+    }
+
+    /**
+     *
+     * @param currentMonitor
+     * @return The height of the screen taking into account display scaling
+     */
+    public static Dimension getScaledScreenSize(Component component) {
+        return getScaledScreenSize(component.getGraphicsConfiguration().getDevice().getDisplayMode());
+    }
+
+    /**
+     *
+     * @param currentMonitor
+     * @return The height of the screen taking into account display scaling
+     */
+    public static Dimension getScaledScreenSize(DisplayMode currentMonitor) {
+        int monitorH = currentMonitor.getHeight();
+        int monitorW = currentMonitor.getWidth();
+        int pixelPerInch= Toolkit.getDefaultToolkit().getScreenResolution();
+        return new Dimension(
+                DEFAULT_DISPLAY_PPI * monitorW / pixelPerInch,
+                DEFAULT_DISPLAY_PPI * monitorH / pixelPerInch);
+    }
+
+    /**
+     *
+     * @param image
+     * @param maxWidth
+     * @param maxHeight
+     * @return an image with the same aspect ratio that fits within the given bounds, or the existing image if it already does
+     */
+    public static Image constrainImageSize(Image image, ImageObserver observer, int maxWidth, int maxHeight) {
+        Image newImage = image;
+        if (newImage.getWidth(observer) > maxWidth) {
+            newImage = newImage.getScaledInstance(maxWidth, -1, Image.SCALE_DEFAULT);

```

Comment:
Im not sure but I think SCALE_SMOOTH or something like that is better than SCALE_DEFAULT. (And with images that have transparency even that isnt good. I think the best image scaler is used for displaying the scaled image icons)

Reply:
That would be ImageUtil.getscaledInstance